### PR TITLE
stop php from sending any Cache control headers

### PR DESCRIPTION
**Pull request by butonic**
## Description

On debian the default setting for session.cache_limiter is nocache causing Pragma, CacheControl and Expiry headers to be added to every response. This may lead to conflicting headers where owncloud sets a Cache-Control header that allows caching but Pragma is still nocache. Pragma is only needed to tell IE8 how to cache responses. No longer needed -> kill it.

This PR changes the default setting of session.cache_limiter to '' to prevent php from sending any caching related headers to get give oc full control and prevent conflicting headers.

see http://php.net/manual/de/function.session-cache-limiter.php#refsect1-function.session-cache-limiter-parameters

## Related Issue

* https://github.com/owncloud/core/pull/26589/files#diff-17d17c6a954f29c12717db921f235531L137
* https://github.com/owncloud/core/issues/21036

## Motivation and Context

cleaner caching, no unnecessary headers, profit?

## How Has This Been Tested?

Firefox, Edge, Chromium

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by PVince81** [comment]
Can anyone familiar with .htaccess and php settings review this ? @DeepDiver1975 @PhilippSchaffrath @felixheidecke

**Comment by felixheidecke** [comment]
Sorry … no xp in caching :-/

File: .htaccess [hunk 1]
```diff
@@ -32,7 +32,8 @@
   php_value mbstring.func_overload 0
   php_value always_populate_raw_post_data -1
   php_value default_charset 'UTF-8'
-  php_value output_buffering 0
+  php_value output_buffering 0s
```
Comment (PhilippSchaffrath) [inline .htaccess]: Documentation says this is either a bool or an integer. Am i missing something @butonic?
http://php.net/manual/de/outcontrol.configuration.php#ini.output-buffering

File: .htaccess [hunk 2]
```diff
@@ -32,7 +32,8 @@
...
+  php_value session.cache_limiter ''
```
Comment (PhilippSchaffrath) [inline .htaccess]: This should be either `public`, `private_no_expire`, `private` or `nocache`.
http://php.net/manual/de/function.session-cache-limiter.php

File: .htaccess [hunk 3]
```diff
@@ -44,6 +45,7 @@
   php_value mbstring.func_overload 0
   php_value default_charset 'UTF-8'
   php_value output_buffering 0
+  php_value session.cache_limiter ''
```
Comment (PhilippSchaffrath) [inline .htaccess]: Same here.

File: .user.ini [hunk 4]
```diff
@@ -5,3 +5,4 @@ mbstring.func_overload=0
 always_populate_raw_post_data=-1
 default_charset='UTF-8'
 output_buffering=0
+session.cache_limiter=''
```
Comment (PhilippSchaffrath) [inline .user.ini]: Same here.

**Review by phisch — CHANGES_REQUESTED** [review]

File: .htaccess (hunk 2 above)
Comment (DeepDiver1975) [inline .htaccess]: '' is valid 
````
Setting the cache limiter to '' will turn off automatic sending of cache headers entirely.

Comment (PhilippSchaffrath) [inline .htaccess]: Oh right.

File: .htaccess (hunk 1 above)
Comment (butonic) [inline .htaccess]: meat fingers ...

**Review by phisch — APPROVED** [review]

**Comment by PhilippSchaffrath** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
- [ ] needs squash on merge

**Comment by PhilippSchaffrath** [comment]
@DeepDiver1975 jenkins build successful, but did not report here, can you re-run the build?

**Comment by butonic** [comment]
cc @VicDeo regarding IE11

**Comment by butonic** [comment]
https://github.com/owncloud/core/issues/26922 has a workaround for IE caching Ajax requests without this caching

**Comment by VicDeo** [comment]
well... this introduces a huge variety of IE-related issues and solves none of them at the same time.
So the only workaround would be to revert this in production :/

`mod_headers` is not always available and apache is not the only web-server. 

I'd say before taking a control on cache one should know what to do with that control.

**Comment by PVince81** [comment]
Seeing the comments from @VicDeo and the fact that you ticked "Breaking change (fix or feature that would cause existing functionality to change)" doesn't inspire trust in this PR. Also I don't have enough experience with caching headers and quirks.

@butonic is this something you'd like to continue to pursue or should we move to backlog or close this PR ?

**Comment by PVince81** [comment]
can we close ?

**Comment by PVince81** [comment]
closing due to lack of feedback

**Pull request closed without merging**